File: java/oop/CityApp/src/com/techlabs/citymap/CityMap.java
package com.techlabs.citymap;

import java.util.HashMap;
import java.util.Map.Entry;

public class CityMap {
	private HashMap<City,City> cityMap=new HashMap<City, City>();
	
	public void addCity(String cityCode,String cityDescription){
		City city=new City(cityCode,cityDescription);
		cityMap.put(city, city);
	}

	public HashMap<City,City> search(String cityCode){
		HashMap<City, City> matchingCity=new HashMap<City, City>();
		
		for(Entry<City, City> city:cityMap.entrySet()){
			if(city.getKey().getCityCode().contains(cityCode.toUpperCase())){
				matchingCity.put(city.getKey(), city.getValue());
			}
		}
		
		return matchingCity;
	}
}

File: dotNet/CSharp/CityApp/CityApp/CityMap.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CityApp
{
    class CityMap
    {
        private Dictionary<City, City> _cityBook = new Dictionary<City, City>();

        public void addCity(string cityCode,string cityDescription)
        {
            City city = new City(cityCode,cityDescription);
            _cityBook.Add(city, city);
        }

        public Dictionary<City,City> search(string cityCode)
        {
            Dictionary<City, City> matchingCity = new Dictionary<City, City>();

            foreach(KeyValuePair<City,City> city in _cityBook)
            {
                if (city.Key.CityCode.Contains(cityCode.ToUpper()))
                {
                    matchingCity.Add(city.Key, city.Value);
                }
            }

            return matchingCity;
        }
    }
}
